src/0_storage.js

```diff
@@ -50,16 +50,28 @@ var webStorage = function(perm) {
 	var storageMethod = perm ? localStorage : sessionStorage;
 	return {
 		getAll: function() {
-			var allKeyValues = { };
+			var allKeyValues = null;
 			for (var key in storageMethod) {
-				if (key.indexOf(BRANCH_KEY_PREFIX) == 0) { allKeyValues[trimPrefix(key)] = retrieveValue(storageMethod.getItem(key)); }
+				if (key.indexOf(BRANCH_KEY_PREFIX) == 0) {
+					// The first check depend on this being false. An empty object is not false
+					if (allKeyValues === null) {
+						allKeyValues = {};
+					}
+					allKeyValues[trimPrefix(key)] = retrieveValue(storageMethod.getItem(key));
+				}
 			}
 			return allKeyValues;
 		},
 		get: function(key) { return retrieveValue(storageMethod.getItem(prefix(key))); },
 		set: function(key, value) { storageMethod.setItem(prefix(key), value); },
 		remove: function(key) { storageMethod.removeItem(prefix(key)); },
-		clear: function() { storageMethod.clear(); },
+		clear: function() {
```

Comment:
The clear function for local storage was wiping out everything so I've made it work based on the key prefix